### ISSUE TITLE
Increase the current limit for attributes from 8 to 12 or 16

### ISSUE THREAD

**Issue by Spiri0** [mrdoob/three.js#28942]
### Description

At the moment I can create a maximum of 8 attributes for custom buffer geometries in the node system. As far as I know, at least 16 are possible for webgl and webgpu.

So this falls into the suggestion area.

I currently use 7 attributes, but could need 2 more, which would bring me to 9, so one more attribute than possible. This is nothing urgent and nothing critical because it's not an error.

However, more than 8 attributes are desirable because I can use attributes to implement advanced morphing techniques and thus save massive amounts of computing effort in the shader.

### Solution

The solution is to increase the limit for attributes from 8 to 16.

### Alternatives

The alternative is that I carry out calculations on the shader side. Of course that works. However, if I can use target attributes calculated once, it is more efficient than over 100 lines of code in the shader that are run through at every interval.

### Additional context

For you as developers so that you can also see a little of what is possible with WebGPU and the node system:

![mars65k](https://github.com/user-attachments/assets/4ed083a5-6cca-4eb5-b74c-3c49de5b69fe)

![mars65k2](https://github.com/user-attachments/assets/5044b2b2-b8f5-4dcc-9300-fb3319c4a810)

![131k_3](https://github.com/user-attachments/assets/4f405ac7-78c6-4f0b-9b3d-0e955775d848)

I still have some work to do but the most difficult part is over.
Let's see if I can put it online this year.

**Comment by Mugen87** [comment]
Can you please demonstrate the attribute limit of 8 with a live example? 

I'm not aware of a hard-coded value limitation in the engine.

**Comment by Spiri0** [comment]
Strange, as a test I have now assigned the same attribute (a Float32BufferAttribute vec4) to a buffer geometry several times under different names. Now I'm already on the 13th attribute and it works without the warning appearing in the console. 
I should have recorded this around three weeks ago when I had the case because now I can't get the case reproduced.
Please leave the point unassigned. When I get it reproduced again I'll add an example. If not, I will close the point in about two weeks because I'm taking care of my issues and I don't want to have anything open that I can't get reproduced again.

By the way, the extension to include the texture_2d_array works very well. I tested that straight away. I think it's very good that Sunag took all texture types into account for the wgslFn and not just the 2D array texture that I wished.

**Comment by aardgoose** [comment]
Note for webGPU there is a default limit of 16 vertex attributes and 8 vertex buffers - so if you use interleaved attributes you can exceed 8 attributes with the default limits. You can request higher limits when you request the gpuDevice instance if supported.

**Comment by Mugen87** [comment]
Closing. We can reopen the topic if the is an actual issue in the engine.

**Comment by Spiri0** [comment]
Ah, that explains a lot. Now I think I understand why I couldn't reproduce this with my attempt yesterday because I always used the same vertex buffer.

But if I want to use 8 completely independent data arrays with attributes, I run into the vertex buffer limit.

Most people will never need that many attributes. Therefore the topic is not important anyway. 

I think I've found an alternative. I have 3 scalar attributes. I replaced this with a vec4 attribute. Instead of three attributes for different scalar data, I now use an attribute vec4 in which the scalar data is in x, y, z and I then have w as a reserve
